libkbfs/cr_chains.go

```diff
@@ -144,18 +145,53 @@ func (cc *crChain) getActionsToMerge(renamer ConflictRenamer, mergedPath path,
 }
 
 func (cc *crChain) isFile() bool {
+	return cc.file
+}
+
+// identifyType figures out whether this chain represents a file or
+// directory.  It tries to figure it out based purely on operation
+// state, but setAttr(mtime) can apply to either type; in that case,
+// we need to fetch the block to figure out the type.
+func (cc *crChain) identifyType(ctx context.Context, fbo *folderBlockOps,
+	md *RootMetadata) error {
 	if len(cc.ops) == 0 {
-		return false
+		return nil
 	}
 
-	// If the first op is setAttr or sync, this is a file chain.
-	switch cc.ops[0].(type) {
-	case *syncOp:
-		return true
-	case *setAttrOp:
-		return true
+	// If any op is setAttr (ex or size) or sync, this is a file
+	// chain.  If it only has a setAttr/mtime, we don't know what it
+	// is, just return false.
```

Comment:
return false -> fall through. Also mention that if there are any other op types, we know it's a dir.